File: templates/java/zh/text/code/pothole.java
import stanford.卡雷尔.*;
public class 坑洞填充卡雷尔 extends 卡雷尔 {
   public void run() {
      move();
      填充坑洼();
      move();
   }

   /**
    * 通过在该角落放置一个锥体填充卡雷尔当前位置下方的坑洞。为了使这种方法正常工作，卡雷尔必须在坑洞正上方朝东。执行完成后，卡雷尔将返回同一个广场并再次面向东方。
    */
   private void 填充坑洼() {
      turnRight();
      move();
      putBeeper();
      turnAround();
      move();
      turnRight();
   } 

   /** 卡雷尔向右转90度。 */
   public void turnRight() {
      turnLeft();
      turnLeft();
      turnLeft();
   }

   /** 将卡雷尔变成180度左右。 */
   public void turnAround() {
      turnLeft();
      turnLeft();
   }
}

File: templates/python/zh/text/code/pothole.py
from 卡雷尔.stanford import *
def main():
   move()
   填充坑洼()
   move()

＃通过以下方式填充Karel当前位置下方的坑洼
＃在该角上放置一个锥体 。为了使此功能起作用
＃正确，卡雷尔必须正对着上方的东方
＃坑洼。执行完成后，Karel将拥有
＃返回同一广场，再次面向东方。
def 填充坑洼():
   turn_right()
   move()
   放锥体()
   turn_around()
   move()
   turn_right()

＃将Karel向右旋转90度。
def turn_right():
   turn_left()
   turn_left()
   turn_left()

＃将Karel旋转180度。
def turn_around():
   turn_left()
   turn_left()